backends-velox/src/main/scala/org/apache/gluten/execution/RowToVeloxColumnarExec.scala

```diff
@@ -145,10 +145,20 @@ object RowToVeloxColumnarExec {
         }
       }
 
-      def nativeConvert(row: UnsafeRow): ColumnarBatch = {
+      def convertToUnsafeRow(row: InternalRow): UnsafeRow = {
+        row match {
+          case unsafeRow: UnsafeRow => unsafeRow
+          case _ =>
+            converter.apply(row)
+        }
+      }
+
+      override def next(): ColumnarBatch = {
+        val firstRow = it.next()
```

Comment:
why the first row needs to be handled seperately?

Reply:
Because we need to get estimate row size by the first row